[IMP] email marketing: update essentials doc (#7521)
odoo/documentation: content/applications/marketing/email_marketing.rst

   :align: center
   :alt: View of the campaign feature setting in Odoo Email Marketing.

Once the :guilabel:`Mailing Campaigns` feature is activated, a new :guilabel:`Campaigns` menu option
appears in the header.

When that is clicked, Odoo reveals a separate :guilabel:`Campaigns` page, displaying all the mailing
campaigns in the database, and the current stage they are in, showcased in a default Kanban view.

.. image:: email_marketing/campaigns-page.png
   :align: center
   :alt: View of the campaign page in Odoo Email Marketing.

.. note::
   This information can also be viewed in a list, by clicking the :guilabel:`☰ (horizontal lines)`
   icon in the upper-right corner.

Clicking any campaign from the :guilabel:`Campaigns` page reveals that campaign's form.

There are two different ways to create and customize campaigns in the *Email Marketing* application,
either directly from the :ref:`Campaigns page <email_marketing/campaign-page>` or through the
:ref:`Settings tab <email_marketing/campaign-settings>` on an email form.

.. _email_marketing/campaign-page:

Create mailing campaign (from campaigns page)
---------------------------------------------

> When the :guilabel:`Mailing Campaigns` feature is activated, a new :guilabel:`Campaigns` option

Reviewer: ```suggestion
When the *Mailing Campaigns* feature is activated, a new *Campaigns* option appears in the header of
the *Email Marketing* application. Campaigns can be created directly on the *Campaigns* page in the
*Email Marketing* app.
```

listed emails. Those columns are as follows:

- :guilabel:`Date`: the date the email was sent.
- :guilabel:`Subject`: the subject of the email.
- :guilabel:`Responsible`: the user who created the email, or the user who has been assigned to the
   email.
- :guilabel:`Sent`: how many times the email has been sent.
- :guilabel:`Delivered (%)`: percentage of sent emails that have been successfully delivered.
- :guilabel:`Opened (%)`: percentage of sent emails that have been opened by the recipients.
- :guilabel:`Clicked (%)`: percentage of sent emails that have been clicked by the recipients.
- :guilabel:`Replied (%)`: percentage of sent emails that have been replied to by the recipients.
- :guilabel:`Status`: the status of the email (:guilabel:`Draft`, :guilabel:`In Queue`, or
  :guilabel:`Sent`).

To add or remove columns, click the :guilabel:`Additional Options (two horizontal lines with dots)`
icon, located to the far-right of the column titles in list view. Doing so reveals a drop-down menu
of additional column options.

.. _email_marketing/kanban-view:

Kanban view
-----------

The Kanban view, represented by the :guilabel:`(inverted bar graph)` icon, can be accessed in the

> upper-right corner of the :guilabel:`Mailings` dashboard in the *Email Marketing* application.

Reviewer: ```suggestion
upper-right corner of the :guilabel:`Mailings` dashboard in the :guilabel:`Email Marketing` app.
```

Author: So, this change is kinda "news" to me...So, even if it just appears in the header as the app title, we should always use a GUI? I wasn't aware of that. I'm implementing all these changes, but just wanna know if this is going to be the standard moving forward.

Author: it will also help me in my Final Reviews to know this, as well. Thanks

Reviewer: Sure! I am happy to discuss this because I want to be on the same page as well! 

The criteria that I look for in order to use a :guilabel: is to ask: is the text on the screen during this active instruction, if so, use a :guilabel: for the text, otherwise italics can be used.

Our [guidelines](https://www.odoo.com/documentation/master/contributing/documentation/rst_cheat_sheet.html#gui-element) state:
> Use the guilabel markup to identify any text of the interactive user interface

So while we do italics on app names (i.e., _Email Marketing_), in this instance on line 86, the text "Email Marketing" is present on the UI in this instruction, so thats why I am suggesting to use a :guilabel:.

Let me know your thoughts, and tagging @StraubCreative here as well so we can discuss. Thanks!

- :guilabel:`Draft`: the email is still being written/created.
- :guilabel:`In Queue`: the email is scheduled to be sent at a later date.
- :guilabel:`Sending`: the email is currently being sent to its recipients.
- :guilabel:`Sent`: the email has already been sent to its recipients.

In each stage, there are drag-and-drop cards representing the emails that have been created/sent,
and the stage they are in represents the current status of that mailing.

Each card on the :guilabel:`Mailings` dashboard provides key information related to that specific
email.

When the cursor hovers over the upper-right corner of an email campaign card, :guilabel:`⋮ (three
vertical dots)` appear. When clicked, a mini drop-down menu reveals the option to color-code the
email, :guilabel:`Delete` the email, or :guilabel:`Archive` the message for potential future use.

.. image:: email_marketing/three-dot-dropdown.png
   :align: center
   :alt: View of the three-dot drop-down menu on the Odoo Email Marketing dashboard.

.. _email_marketing/calendar-view:

Calendar view
-------------

The calendar view, represented by a :guilabel:`📆 (calendar)` icon, can be accessed in the

> upper-right corner of the :guilabel:`Mailings` dashboard in the *Email Marketing* app.

Reviewer: ```suggestion
upper-right corner of the :guilabel:`Mailings` dashboard in the :guilabel:`Email Marketing` app.
```

While in calendar view, a monthly calendar (by default), shows when the mailings have been sent or
are scheduled to be sent.

.. image:: email_marketing/calendar-view.png
   :align: center
   :alt: Calendar view of the mailings dashboard in the Email Marketing application.

The current date is represented by a :guilabel:`🔴 (red circle)` icon over the date on the calendar.

To the right of the calendar, the options to filter the results by :guilabel:`Responsible` and/or
:guilabel:`Status` are available, via checkboxes.

.. tip::

>    To hide the right sidebar, click the :guilabel:`(white and black box)` icon, located above the

Reviewer 2: This Odoo UI icon is called `panel-right`
![Screenshot 2024-02-23 at 12 03 01 PM](https://github.com/odoo/documentation/assets/5496809/b7ff7da3-cbcf-4b91-af67-de0ad6749383)

```suggestion
   To hide the right sidebar, click the :guilabel:`(panel-right)` icon, located above the
```

   sidebar.

In the top-left corner, above the calendar, the option to change the time period being displayed is
available via a drop-down menu, which shows :guilabel:`Month`, by default. When clicked, the
drop-down menu that appears reveals the options: :guilabel:`Day`, :guilabel:`Week`,
:guilabel:`Month` (default), :guilabel:`Year`, and :guilabel:`Show weekends` (selected by default).

Clicking any of those options changes the calendar display to reflect that desired amount of time.

Clicking either :guilabel:`⬅️ (left arrow)` icon or :guilabel:`➡️ (right arrow)` icon changes the
calendar to a previous or future time, depending on what is clicked, based on the chosen amount of
time being represented.

To jump back to the current date, click the :guilabel:`Today` button.

.. _email_marketing/graph-view:

Graph view
----------

The graph view, represented by a :guilabel:`(line graph)` icon, can be accessed in the upper-right

> corner of the :guilabel:`Mailings` dashboard in the *Email Marketing* app.

Reviewer: ```suggestion
corner of the :guilabel:`Mailings` dashboard in the :guilabel:`Email Marketing` app.
```

.. image:: email_marketing/graph-view.png
   :align: center
   :alt: How the Graph view appears in the Odoo Email Marketing application.

In the upper-left corner, above the graph, there is a :guilabel:`Measures` drop-down menu. When
clicked, different filter options become available to further customize the graph views.

Those :guilabel:`Measures` options are: :guilabel:`A/B Testing percentage` and :guilabel:`Count`
(default).

To the right of the :guilabel:`Measures` drop-down menu is an :guilabel:`Insert in Spreadsheet`
button, if the *Documents* application is installed. When clicked, a pop-up window appears, in which
the ability to add the graph to a spreadsheet or dashboard becomes available.

Beside the :guilabel:`Measures` drop-down menu and :guilabel:`Insert in Spreadsheet` button are
different graph view options. From left-to-right, those graph view options are: :guilabel:`(bar
chart)` (default), :guilabel:`(line chart)`, and :guilabel:`(pie chart)`.

.. note::
   Each graph view option provides its own series of additional view options, which appear to the
   right of the selected graph view option.

Search options
--------------

> Regardless of the view chosen for the :guilabel:`Mailings` dashboard in the *Email Marketing* app,
> the :guilabel:`Filters`, :guilabel:`Group by`, and :guilabel:`Favorites` options are always
> available to further customize the information being displayed.

Reviewer: ```suggestion
Regardless of the view chosen for the :guilabel:`Mailings` dashboard in the :guilabel:`Email
Marketing` app, the :guilabel:`Filters`, :guilabel:`Group by`, and :guilabel:`Favorites` options are
always available to further customize the information being displayed.
```

To access those options, click the :guilabel:`(downward arrow)` icon, located to the right of the
search bar. Doing so reveals a drop-down mega menu featuring those filtering and grouping options.

.. image:: email_marketing/search-mega-menu.png
   :align: center
   :alt: The drop-down mega menu of search options the Odoo Email Marketing application.

These options provide various ways to specify and organize the information seen on the
:guilabel:`Mailings` dashboard.

.. tabs::

   .. tab:: Filters

      This section of the drop-down mega menu provides different ways to filter email results being

>       shown on the :guilabel:`Mailings` dashboard in the *Email Marketing* app.

Reviewer: ```suggestion
      shown on the :guilabel:`Mailings` dashboard in the :guilabel:`Email Marketing` app.
```

      .. image:: email_marketing/filters-dropdown.png
         :align: center
         :alt: View of filters drop-down menu options on the Odoo Email Marketing dashboard.

      The options are: :guilabel:`My Mailings`, :guilabel:`Sent Date`, :guilabel:`A/B Tests`,
      :guilabel:`A/B Tests to review`, :guilabel:`Archived`, and :guilabel:`Add Custom Filter`.

      If :guilabel:`Add Custom Filter` is selected, Odoo reveals a pop-up window, with three
      customizable fields to fill in, in order to create custom filter rules for Odoo to use to
      retrieve results that fit more specific criteria.

      .. image:: email_marketing/add-custom-filter-popup.png
         :align: center
         :alt: Add custom filter pop-up window that appears in Odoo Email Marketing app.

   .. tab:: Group By

      This section of the drop-down mega menu provides different ways to group email results being

>       shown on the :guilabel:`Mailings` dashboard in the *Email Marketing* app.

Reviewer: ```suggestion
      shown on the :guilabel:`Mailings` dashboard in the :guilabel:`Email Marketing` app.
```

selected, but clicking the field reveals a drop-down menu of other recipient options.

With the default :guilabel:`Mailing List` option selected, a specific mailing list **must** be
chosen from the adjacent :guilabel:`Select mailing lists` field drop-down menu.

.. tip::
   More than one mailing list can be chosen from the :guilabel:`Select mailing lists` field.

Odoo then sends the email to contacts on that specific mailing list(s).

.. seealso::
   :doc:`email_marketing/mailing_lists`

When the :guilabel:`Recipients` field is clicked, a drop-down menu of other options is revealed.
Each option provides different ways Odoo can create a target audience for the email.

.. image:: email_marketing/recipients-dropdown.png
   :align: center
   :alt: View of recipients drop-down menu in the Odoo Email Marketing application.

Those options (excluding the default :guilabel:`Mailing List`) provide the option to create a more
specified recipient filter, in an equation-like format, which appears beneath the
:guilabel:`Recipients` field.

The :guilabel:`Recipients` field options, other than the default :guilabel:`Mailing List` option,

> are as follows:

Reviewer 2: The descriptions for these list items are a borderline redundant. Would expand them so they're more specific to the models they're attached to if you can. 

For example: 
- **Lead/Opportunity** ties specifically to records in the _CRM_ application which opens up a number of opportunities to influence sales or purchase decisions (like what we do internally for lost lead reactivation drip campaigns).
- Event Registration ties specifically to the _Events_ app and provides opportunities to interact with event registrants to communicate important information about the event(s) or nurture other valuable actions such as post-event surveys or purchases, etc.

What is the difference between **Contact** and **Mailing Contact** ?

- :guilabel:`Contact`: focuses on specific contacts in the database.
- :guilabel:`Event Registration`: focuses on people in the database who purchased event
  registrations.
- :guilabel:`Lead/Opportunity`: focuses on leads or opportunities in the database.
- :guilabel:`Mailing Contact`: focuses on specific mailing contacts in the database.
- :guilabel:`Sales Order`: focuses on a specific sales orders in the database.

Add recipient filter
~~~~~~~~~~~~~~~~~~~~

To add a more specific recipient filter to any :guilabel:`Recipient` option, select any recipient
option (other than :guilabel:`Mailing List`), and click the :guilabel:`Modify filter (right-facing
arrow)` icon beneath the :guilabel:`Recipient` field to reveal three subsequent filter rule fields,

> formatted like an equation.

Reviewer 2: The framing of this feature is *technically* correct but it seems to lack real-world application knowledge, specifically:

- It is *highly encouraged* to use detailed targeting criteria for the **Recipients** field. It's not a "nice-to-have" or an optional feature— it's very necessary, and one line of targeting logic typically is not enough to run a smart email campaign.
- **Mailing List** is indeed a fair **Recipients** option to choose from as a base but it's not really the standard (not unless you're buying marketing lists and important them into Odoo, which we'd frown upon anyway). The more standard **Recipient** sources would be **Lead/Opportunity** or **Event Registration**, to which, much more detailed targeting criteria would be added on top of those seed sources.

Here's an example of how detailed our targeting should should be in this section
![Screenshot 2024-02-23 at 12 28 35 PM](https://github.com/odoo/documentation/assets/5496809/6c6f9f89-0559-4269-b57c-0badac868bf7)

To reveal the sub-menu options within the filter rule fields, click each field, and make the desired
selections, until the preferred configuration has been achieved.

The number of :guilabel:`records` in the database that match the configured rule(s) are indicated
beneath the configured filter rule(s), in green.

> .. image:: email_marketing/filter-records.png

Reviewer 2: Similar to the comment above. This is not a great example of how to use filters in a real-world setting.

Author: Since I added the above screenshot in an example admonition, I think this can still serve a purpose -- not as a 'real world' setting example, but more of how the rule lines appear on the U/I, and how the 'Records' look on screen, as well. However, if you feel like this screenshot still isn't useful, just let me know. Thanks.

   :align: center
   :alt: View of how recipient filters can be customized in Odoo Email Marketing.

.. note::
   Some sub-menu options in the first rule field allow for a second choice to provide even more
   specificity.

> To the right of each rule, there are :guilabel:`➕ (Add node)`, :guilabel:`(Add branch)`, and
> :guilabel:`🗑️ (Delete node)` icons.
> 
> The :guilabel:`➕ (Add node)` icon adds a node (line) to the rule.
> 
> The :guilabel:`(Add branch)` icon adds a branch to the node. A branch means two additional, indented
> sub-nodes are added to the rule, providing even more specificity to the line above it.
> 
> The :guilabel:`🗑️ (Delete node)` icon deletes a specific node (line) of the rule.

Reviewer 2: Would turn this into a bullet list 👍 
The emoji labels are incorrect, as well; what we write in parenthesis is a fail-safe label in case the emoji fails to load. So `➕ (Add node)` is actually `➕ (plus sign)`. I touched up the definitions slightly, for your consideration.

```suggestion
To the right of each rule, there are three additional options, represented by :guilabel:`➕ (plus
sign)`, :guilabel:`(sitemap)`, and :guilabel:`🗑️ (trash)` icons.

- The :guilabel:`➕ (plus sign)` icon adds a new node (line) to the overall targeting logic.
- The :guilabel:`(sitemap)` icon adds a branch to the node. A branch contains two additional,
  indented sub-nodes that are related to that specific rule, providing even more specificity to the
  parent line above it.
- The :guilabel:`🗑️ (trash)` icon deletes a specific node (line) in the array of logic
```

  "winning" version.
- :guilabel:`Leads`: the mailing with the most leads generated is determined to be the "winning"
  version.
- :guilabel:`Quotations`: the mailing with the most quotations generated is determined to be the
  "winning" version.
- :guilabel:`Revenues`: the mailing with the most revenue generated is determined to be the
  "winning" version.

The :guilabel:`Send Final On` field allows users to choose a date that is used to know *when* Odoo
should determine the "winning" email, and subsequently, send that version of the email to the
remaining recipients.

.. image:: email_marketing/ab-test-tab.png
   :align: center
   :alt: View of the A/B Tests tab in Odoo Email Marketing application.

To the right of those fields is a :guilabel:`Create an Alternative Version` button. When clicked,
Odoo presents a new :guilabel:`Mail Body` tab for the user to create an alternate version of the
email to test.

.. _email_marketing/settings_tab:

Settings tab
------------

The options present in the :guilabel:`Settings` tab of the mail form are divided into two sections:
:guilabel:`Email Content` and :guilabel:`Tracking`.

.. note::

>    The options available in the :guilabel:`Settings` tab vary depending on if the :guilabel:`Mailing
>    Campaigns` feature is activated in the *Settings* page of the *Email Marketing* application
>    (:menuselection:`Email Marketing --> Configuration --> Settings`).

Reviewer: No :guilabel: on "Mailing Campaigns" because the field is not on the screen yet. Can also simplify the navigation instructions and link to the **Mailing Campaigns** section :) 
```suggestion
   The options available in the :guilabel:`Settings` tab vary depending on if the *Mailing
   Campaigns* feature is activated in :menuselection:`Email Marketing --> Configuration -->
   Settings`. See :ref:`email_marketing/mailing-campaigns` for more information.
```

> Without the :guilabel:`Mailing Campaigns` feature activated, the :guilabel:`Settings` tab on the

Reviewer: Here too
```suggestion
Without the *Mailing Campaigns* feature activated, the :guilabel:`Settings` tab on the email form
```

:guilabel:`Attachments`, and :guilabel:`Responsible` fields.

.. image:: email_marketing/settings-without-features.png
   :align: center
   :alt: View of settings tab in Odoo Email Marketing app, without campaign setting activated.

Email content
~~~~~~~~~~~~~

- :guilabel:`Preview Text`: allows the user to enter a preview sentence to encourage recipients to
  open the email. In most inboxes, this is displayed next to the subject. If left empty, the first
  characters of the email content appear, instead. The ability to add an emoji in this field is
  available, as well, via the :guilabel:`(smiley face with a plus sign)` icon.
- :guilabel:`Send From`: designate an email alias that displays as the sender of this particular
  email.
- :guilabel:`Reply To`: designate an email alias to whom all the replies of this particular email
  are sent.
- :guilabel:`Attach a file`: if any specific files are required (or helpful) for this email, click
  the :guilabel:`Attachments` button, and upload the desired file(s) to the email.

Tracking
~~~~~~~~

- :guilabel:`Responsible`: designate a user in the database to be responsible for this particular
  email.

.. note::

>    If the :guilabel:`Mailing Campaign` feature *is* activated, an additional :guilabel:`Campaign`
>    field appears in the :guilabel:`Tracking` section of the :guilabel:`Settings` tab.

Reviewer: ```suggestion
   If the *Mailing Campaign* feature *is* activated, an additional :guilabel:`Campaign` field
   appears in the :guilabel:`Tracking` section of the :guilabel:`Settings` tab.
```

   .. image:: email_marketing/settings-tab-with-campaign.png
      :align: center
      :alt: View of settings tab in Odoo Email Marketing when campaign setting is activated.

   The additional :guilabel:`Campaign` field allows users to attach this particular email to a
   mailing campaign, if desired.

   If the desired campaign is not available in the initial drop-down menu, select :guilabel:`Search
   More` to reveal a complete list of all mailing campaigns in the database.

>    Or, type the name of the desired mailing campaign in the :guilabel:`Mailing Campaign` field,
>    until Odoo reveals the desired campaign in the drop-down menu. Then, select the desired campaign.

Reviewer: Double checking– is this supposed to be the **Campaign** field or **Mailing Campaign** field?
```suggestion
   Or, type the name of the desired mailing campaign in the :guilabel:`Campaign` field, until Odoo
   reveals the desired campaign in the drop-down menu. Then, select the desired campaign.

```

In this pop-up window, click the :guilabel:`Send on` field to reveal a calendar pop-up window.

.. image:: email_marketing/schedule-popup-calendar.png
   :align: center
   :alt: View of pop-up window that appears when the schedule button on an email form is clicked.

From the calendar pop-up window, select the future date and time for Odoo to send this email. Then,
click :guilabel:`✔️ Apply`. When a date and time are chosen, click the :guilabel:`Schedule` button,
and the status of the mailing changes to :guilabel:`In Queue`.

.. _email_marketing/test:

Test
----

The :guilabel:`Test` button reveals a :guilabel:`Test Mailing` pop-up window.

.. image:: email_marketing/test-popup.png
   :align: center
   :alt: View of pop-up window that appears when the test button on an email form is clicked.

From this pop-up window, enter the email addresses of the contacts to whom Odoo should send this
test email in the :guilabel:`Recipients` field. Multiple contacts can be added in this field, if
desired.

Once all the desired email addresses have been entered in the :guilabel:`Recipients` field, click
the :guilabel:`Send Test` button.

.. warning::

>    By default, there's a daily limit applied for **all emails** sent throughout
>    **all applications**. So, if there are remaining emails to be sent after a limit has been
>    reached, those mailings are **not** sent automatically the next day. The sending needs to be
>    forced, by opening the email and clicking :guilabel:`Retry`.

Reviewer: Early break
```suggestion
   By default, there's a daily limit applied for **all emails** sent throughout **all
   applications**. So, if there are remaining emails to be sent after a limit has been reached,
   those mailings are **not** sent automatically the next day. The sending needs to be forced, by
   opening the email and clicking :guilabel:`Retry`.
```

> 

Reviewer: Can add this reference if using the previous suggestion :)
```suggestion
.. _email_marketing/mailing-campaigns:

```
